app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Admin::UsersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :authorize_admin
+  load_and_authorize_resource
+  before_action :set_user, only: [ :update ]
+
+  # PATCH admin/users/:id
+  def update
+    if @user.update(admin_user_params)
+      render json: { message: "User updated successfully!" }
+    else
+      render json: { errors: @user.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def authorize_admin
+    render json: { error: "You are not authorized to perform this action" }, status: :forbidden unless current_user.role == "admin"
+  end
```

Comment:
❕ Do you have an `AdminController`? I think it would be worth making. It should inherit from `ApplicationController` and then these namespaced `Admin::<somethingelse>` controllers inherit from `AdminController`. Then you can add this method there! You will likely end up with other admin-only controllers